tools/tpm2_import.c

```diff
@@ -176,7 +178,15 @@ static bool encrypt_seed_with_tpm2_rsa_public_key(void) {
         LOG_ERR("BN_bin2bn failed\n");
         goto error;
     }
+#if OPENSSL_VERSION_NUMBER < 0x1010000fL /* OpenSSL 1.1.0 */
```

Comment:
The `ssl_RSA_set0_key` does currently not return an error code, but it does have an error return path. Should this path be taken for some reason then we'll end up without a valid `rsa` structure and we'll leak `n`.

You could make it a real drop-in replacement (i.e. rename `ssl_RSA_set0_key` to `RSA_set0_key` and add the return code handling). This would also get rid of the additional #if/#else preprocessor branch.

Reply:
@mgerstner indeed, I've created PR #564 with your suggested cleanup.